cardano/src/block/block.rs

```diff
@@ -361,11 +361,11 @@ impl chain_core::property::Deserialize for Block {
 }
 
 impl chain_core::property::HasTransaction for Block {
-    type Transactions = [TxAux];
-    fn transactions(&self) -> &Self::Transactions {
+    type Transaction = TxAux;
+    fn transactions<'a>(&'a self) -> Box<Iterator<Item = &Self::Transaction> + 'a> {
```

Comment:
This introduces allocations and dynamic dispatch in iterator loops for the "production" blockchain. I'd like to avoid it if possible.